test/modules/TestResultCoordinator/CountingReportGenerator.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Storage;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    sealed class CountingReportGenerator : ITestResultReportGenerator
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger(nameof(CountingReportGenerator));
+
+        readonly string expectedSource;
+        readonly string actualSource;
+        readonly ISequentialStore<TestOperationResult> expectedStore;
+        readonly ISequentialStore<TestOperationResult> actualStore;
+        readonly string resultType;
+        readonly ITestResultComparer<TestOperationResult> testResultComparer;
+        readonly int batchSize;
+
+        CountingReport<TestOperationResult> report;
+
+        public CountingReportGenerator(
+            string expectedSource,
+            ISequentialStore<TestOperationResult> expectedStore,
+            string actualSource,
+            ISequentialStore<TestOperationResult> actualStore,
+            string resultType,
+            ITestResultComparer<TestOperationResult> testResultComparer,
+            int batchSize = 500)
+        {
+            this.expectedSource = Preconditions.CheckNonWhiteSpace(expectedSource, nameof(expectedSource));
+            this.expectedStore = Preconditions.CheckNotNull(expectedStore, nameof(expectedStore));
+            this.actualSource = Preconditions.CheckNonWhiteSpace(actualSource, nameof(actualSource));
+            this.actualStore = Preconditions.CheckNotNull(actualStore, nameof(actualStore));
+            this.resultType = Preconditions.CheckNonWhiteSpace(resultType, nameof(resultType));
+            this.testResultComparer = Preconditions.CheckNotNull(testResultComparer, nameof(testResultComparer));
+            this.batchSize = batchSize;
+        }
+
+        public string ActualSource => this.actualSource;
+
+        public ISequentialStore<TestOperationResult> AcutalStore => this.actualStore;
+
+        public string ExpectedSource => this.expectedSource;
```

Comment:
Why are all these being exposed as public properties? Where are they being used?

Reply:
changed to internal; used in tests.